fourkitchens/emulsify
# 115: ES6 support, ESLint, Babel 7 Upgrade and Babel Polyfill

## [115: Add ESlint for JS](https://github.com/fourkitchens/emulsify/issues/115)
_This PR changed from addressing merely ESLint to moving to ES6 syntax support for all default code._

### Companion PR in Emulsify Gulp https://github.com/fourkitchens/emulsify-gulp/pull/108. Must be merged/deployed at the same time.

**Changes/additions**:
- Updates to Babel 7 (Emulsify Gulp PR)
- Adds ESLint to deps (Emulsify Gulp PR)
- Adds ESLint config files (ignore and rc)
- Adds Babel config file
- Adds Babel polyfill (since we/Drupal support IE11 out of the box)
- Updates script files to ES6 Syntax! 

**To Test**:
- [ ] Checkout this branch in your local Emulsify repo
- [ ]  If you can use `npm link` to test the Emulsify Gulp PR (I couldn't get it to work), make sure you've done that and checked out the `eslint-dep` branch there. If it doesn't work for you either, replace `^4.0.0` with `fourkitchens/emulsify-gulp#eslint-dep` in [this line](https://github.com/fourkitchens/emulsify/blob/develop/package.json#L6) of package.json (please don't commit the change) and run `yarn`. 
- [ ] Run `yarn start` and ensure there are no errors and that the JS components ([accordion](http://localhost:3000/pattern-lab/public/?p=organisms-accordion), [main menu (mobile)](http://localhost:3000/pattern-lab/public/?p=molecules-main-menu) and [tabs](http://localhost:3000/pattern-lab/public/?p=molecules-tabs)) work correctly in all browsers including IE11
- Ensure no JS console errors
- Verify code looks up to snuff
